#2642 fixed creating alphanumeric data point in virtual data source in new UI

Patrykb0802 opened the pull request:
Fixed creating Alphanumeric data point in Virtual Data Source in new UI

Problem was that some values for points were not flagged as essential and were causing crashes when left empty

Added RuleNotNull for textareas that were causing crashes on creating new data points when left empty

Added two functions handling default values that are being displayed in data point creation window: handleDefaultChangeType and handleNoChangeStartValue

Patrykb0802 committed:
#2642 Fixed creating Alphanumeric data point in Virtual Data Source in new UI

Problem was when you select Change Type e.g. Increment and you change Point Type from for example Numeric to Alphanumeric, where Numeric supports Increment and Alphanumeric doesn't, Change Type droplist is being left empty, than you can press Create/Update button and error occurs.

- Added static (Object.freeze()) defaultSelectedChangeType variable to set it everytime we create/modify data point inside point.vue, which is default value for Change Type droplist now.

- Set defaultSelectedChangeType as v-model for v-select

Patrykb0802 committed:
#2642 Fixed creating Alphanumeric data point in Virtual Data Source in new UI

Problem was that some values for points were not flagged as essential and were causing crashes when left empty

- Added RuleNotNull for textareas that were causing crashes on creating new data points when left empty

- Added two functions handling default values that are being displayed in data point creation window: handleDefaultChangeType and handleNoChangeStartValue

Limraj reviewed (commented):
Here it is easier to correct it, change the default type values in:
..Scada-LTSscadalts-ui/src/components/datasources/VirtualDataSource/VirtualDataPoint.js

On the other hand, requiring values ​​to be set in fields is a good direction.

Patrykb0802 committed:
#2642 Fixed creating Alphanumeric data point in Virtual Data Source in new UI

- Added watch attribute inside VirtualDataSource\point.vue to handle the default value of changeTypeId and noChange.startValue

- Changed default value of datapoint's changeTypeId to NoChange inside DataPointMock.js and VirtualDataPoint.js

Patrykb0802 committed:
#2642 Fixed creating Alphanumeric data point in Virtual Data Source in new UI

- Added resetPointLocatorToDefault() function that resets datapoint to default inside on data type change.

- Added ruleSelectNotNull rule in computed properties to handle necessary to fill in dropdown lists
